### PR TITLE
improve kic engine

### PR DESCRIPTION
**Pull request by medyagh**
- refactor kic, get rid of node package, remove unused funcs, unexport things https://github.com/kubernetes/minikube/issues/6396 
- detecet all containers created by kic using label (for better clean up)
- be able to clean up kic even if user deleted profile and machine folders.
- fix kic restart problem: https://github.com/kubernetes/minikube/issues/6443
- fix kic minikube status warnining https://github.com/kubernetes/minikube/issues/6410
closes https://github.com/kubernetes/minikube/issues/6383  
closes https://github.com/kubernetes/minikube/issues/6410
closes https://github.com/kubernetes/minikube/issues/6443
closes https://github.com/kubernetes/minikube/issues/6396

This PR is in a mergable state, please feel free to merge in my absense if it looks good to you.

### CODE REVIEW

File: pkg/drivers/kic/types.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	StorePath     string            // libmachine store path
+	OCIBinary     string            // oci tool to use (docker, podman,...)
+	ImageDigest   string            // image name with sha to use for the node
+	Mounts        []oci.Mount       // mounts
+	APIServerPort int               // kubernetes api server port inside the container
+	PortMappings  []oci.PortMapping // container port mappings
+	Envs          map[string]string // key,value of environment variables passed to the node
+}
+
+type createConfig struct {
```
Comment (sharifelgamal) [inline pkg/drivers/kic/types.go]: createConfig is a strange name for what this is doing, it feels more like a function name. nodeParams maybe?

File: pkg/minikube/config/profile.go [hunk 2]
```diff
@@ -165,6 +174,26 @@ func ListProfiles(miniHome ...string) (validPs []*Profile, inValidPs []*Profile,
 	return validPs, inValidPs, nil
 }
 
+// removeDups removes duplicates
+func removeDups(profiles []string) []string {
```
Comment (sharifelgamal) [inline pkg/minikube/config/profile.go]: removeDupes or removeDuplicates

**Comment by medyagh** [comment]
created an issue for the cache integeration test failure https://github.com/kubernetes/minikube/issues/6445

File: pkg/minikube/config/profile.go [hunk 3]
```diff
@@ -165,6 +174,26 @@ func ListProfiles(miniHome ...string) (validPs []*Profile, inValidPs []*Profile,
 	return validPs, inValidPs, nil
 }
 
+// removeDupes removes duplicates
```
Comment (tstromberg) [inline pkg/minikube/config/profile.go]: This is simple enough to just inline.

File: pkg/minikube/config/profile.go [hunk 4]
```diff
@@ -165,6 +174,26 @@ func ListProfiles(miniHome ...string) (validPs []*Profile, inValidPs []*Profile,
...
+func removeDupes(profiles []string) []string {
+	// Use map to record duplicates as we find them.
+	seen := map[string]bool{}
+	result := []string{}
+
+	for n := range profiles {
+		if seen[profiles[n]] {
+			// Do not add duplicate.
```
Comment (tstromberg) [inline pkg/minikube/config/profile.go]: instead of an else, use a continue

**Pull request merged**